hyprpm/src/core/PluginManager.cpp

```diff
@@ -398,7 +401,7 @@ bool CPluginManager::updateHeaders(bool force) {
 
     progress.printMessageAbove(std::string{Colors::YELLOW} + "!" + Colors::RESET + " Cloning https://github.com/hyprwm/hyprland, this might take a moment.");
 
-    std::string ret = execAndGet("cd /tmp/hyprpm && git clone --recursive https://github.com/hyprwm/hyprland hyprland");
+    std::string ret = execAndGet("cd /tmp/hyprpm && git clone --recursive https://github.com/hyprwm/hyprland hyprland --shallow-since='" + HLVER.date + "'");
```

Comment:
this won't work, unfortunately. A bunch of cases will violate this:
 - packagers might compile a release days or weeks later
 - source-based compiles still may use releases (gentoo etc)
 
 In these cases, the date will be too far ahead to clone their release

Reply:
The date returned by `hyprctl version` is the date that the commit was made, not the date of compilation. Therefore the clone operation would work regardless of when a release is build.

https://github.com/hyprwm/Hyprland/blob/303b9956b2ae15508b09dffae602550ca17e6539/src/debug/HyprCtl.cpp#L814-L815

Please correct me if I am wrong.

Reply:
right, it is. My bad.